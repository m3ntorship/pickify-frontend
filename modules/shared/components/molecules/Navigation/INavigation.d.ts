import type { ReactElement } from 'react';

declare namespace INavigation {
  export interface IProps {
    profilePic?: string;
  }

  export interface INavLinks {
    name: string;
    path: string;
    content: ReactElement;
  }
}
export { INavigation };
