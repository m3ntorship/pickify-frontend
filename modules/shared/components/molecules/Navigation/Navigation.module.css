.navigation-wrapper {
  @apply p-4 md:px-6 bg-white shadow-navshd flex justify-center fixed z-20 w-full top-0;
}

.navigation {
  @apply w-39xl md:w-60xl flex justify-between items-center;
}

.nav-links ul {
  @apply flex items-center;
}

.nav-links li {
  @apply mr-2 md:mr-6 last:mr-0;
}

.nav-links li.nav-user {
  @apply md:mr-0;
}
