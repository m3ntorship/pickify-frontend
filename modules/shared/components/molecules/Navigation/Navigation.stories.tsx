import React from 'react';
import type { Story, Meta } from '@storybook/react';
import Navigation from './Navigation';
import type { INavigation } from './INavigation';

export default {
  title: 'Molecules/Navigation',
  component: Navigation,
} as Meta;

const Template: Story<INavigation.IProps> = (args) => <Navigation {...args} />;

export const navigation = Template.bind({});
