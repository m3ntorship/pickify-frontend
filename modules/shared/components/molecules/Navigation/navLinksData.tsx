import React from 'react';
import HomeIcon from '../../icons/home.svg';
import FriendsIcon from '../../icons/friends.svg';
import BillIcon from '../../icons/bill.svg';
import HelpIcon from '../../icons/help.svg';
import HappyIcon from '../../icons/happy.svg';
import type { INavigation } from './INavigation';

const isActiveIcon = (path: string, currentPath: string): string =>
  path === currentPath ? 'fill-dark' : 'hover:fill-dark fill-grey';

export const getHomeNavLinks = (pathname: string): INavigation.INavLinks[] => [
  {
    name: 'home',
    path: '/',
    content: <HomeIcon className={isActiveIcon('/', pathname)} />,
  },
  {
    name: 'friends',
    path: '/friends',
    content: <FriendsIcon className={isActiveIcon('/friends', pathname)} />,
  },
  {
    name: 'bill',
    path: '/notifications',
    content: <BillIcon className={isActiveIcon('/notifications', pathname)} />,
  },
];

export const getUserNavLinks = (): INavigation.INavLinks[] => [
  {
    name: 'help',
    path: '/',
    content: <HelpIcon className="fill-grey" />,
  },
  {
    name: 'happy',
    path: '/',
    content: <HappyIcon />,
  },
];
