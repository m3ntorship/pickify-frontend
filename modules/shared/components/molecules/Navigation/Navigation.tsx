import React from 'react';
import type { FC, ReactElement } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { toast } from 'react-toastify';
import styles from './Navigation.module.css';
import Logo from '../../icons/logo.svg';
import Avatar from '../../atoms/Avatar/Avatar';
import Divider from '../../atoms/Divider/Divider';
// import MenuIcon from '../../icons/menu.svg';
import { DividerType } from '../../atoms/Divider/types/EDivider';
import { logoutUser } from '../../../../../context/AuthUserContext/api/authApi';
import { useRedirect } from '../../../hooks/useRedirect/useRedirect';
import { useAuth } from '../../../../../context/AuthUserContext/AuthUserContext';
import DropDown from '../../atoms/DropDown/DropDown';
import TextInput from '../../atoms/TextInputs/TextInput';
import * as ETextInput from '../../atoms/TextInputs/types/ETextInput';
import HelpIcon from '../../icons/help.svg';

import { getHomeNavLinks } from './navLinksData';
import Feedback from '../../organisms/Feedback/Feedback';

const Navigation: FC = (): ReactElement => {
  const { pathname } = useRouter();
  const { user } = useAuth();
  const { redirectToLoginPage, redirectToProfilePage, redirectToFriendsPage } =
    useRedirect();
  const homeNavLinks = getHomeNavLinks(pathname);

  const onMenuClick = async (menuId: string): Promise<void> => {
    switch (menuId) {
      case 'logout':
        try {
          await logoutUser();
          redirectToLoginPage();
        } catch (err: unknown) {
          toast.error('Logging out access is denied');
        }
        break;
      case 'profile':
        redirectToProfilePage();
        break;
      case 'friends':
        redirectToFriendsPage();
        break;
      default:
        toast.error('cant find your option :(');
    }
  };
  const avatarVariant = user?.userImg ? 'filled' : 'notFilled';

  return (
    <nav className={styles['navigation-wrapper']}>
      <div className={styles.navigation}>
        <div className="flex">
          <Link href="/">
            <a className="self-center">
              <Logo className="transform scale-80 -ml-sx" />
            </a>
          </Link>
          <div className={styles['search-box']}>
            <TextInput
              disabled
              inputType={ETextInput.InputType.Search}
              variants={ETextInput.Variants.Default}
              id=""
              value=""
              placeholder="Search Pickify"
              onBlurInputHandler={(): boolean => true}
              onChangeInputValueHandler={(): boolean => true}
              onClickDeleteInputValueHandler={(): boolean => true}
            />
          </div>
        </div>
        <div className={styles['nav-links']}>
          <ul>
            {homeNavLinks.map((homeNavItem) => {
              if (homeNavItem.name === 'friends') {
                return (
                  <li key={homeNavItem.name} className="hidden sm:inline-block">
                    <Link href={homeNavItem.path}>
                      <a>{homeNavItem.content}</a>
                    </Link>
                  </li>
                );
              }
              return (
                <li key={homeNavItem.name}>
                  <Link href={homeNavItem.path}>
                    <a>{homeNavItem.content}</a>
                  </Link>
                </li>
              );
            })}
            <li className="hidden md:inline-block">
              <Divider length="16px" type={DividerType.Vertical} />
            </li>
            <li className="hidden md:inline-block">
              <a role="button" aria-hidden="true">
                <HelpIcon className="fill-grey" />
              </a>
            </li>
            <li className="">
              <Feedback />
            </li>
            <li className={styles['nav-user']}>
              <DropDown
                options={[
                  { id: 'profile', body: 'Profile' },
                  { id: 'friends', body: 'Friends' },
                  { id: 'logout', body: 'Log Out' },
                ]}
                variant="post"
                size="sm"
                onOptionMenuClick={onMenuClick}
              >
                <Avatar
                  size="extra-small"
                  variant={avatarVariant}
                  profilePic={user?.userImg ?? ''}
                />
              </DropDown>
            </li>
            {/* <li className="hidden">
              <MenuIcon />
            </li> */}
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Navigation;
