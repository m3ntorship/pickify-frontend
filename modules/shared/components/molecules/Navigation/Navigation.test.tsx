import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

describe('Navigation', () => {
  it('should render Navigation component with notFilled avatar and Vertical divider', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
