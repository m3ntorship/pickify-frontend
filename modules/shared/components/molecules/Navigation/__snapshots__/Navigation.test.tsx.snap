// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render Navigation component with filled avatar and Vertical divider 1`] = `
<nav
  className="navigation"
>
  <div
    className="links-container"
  >
    <h1
      className="brand-name"
    >
      Pickify
    </h1>
    <ul>
      <li>
        <svg
          data-file-name="SvgHome"
        />
      </li>
      <li>
        <svg
          data-file-name="SvgFriends"
        />
      </li>
      <li>
        <svg
          data-file-name="SvgBill"
        />
      </li>
      <li>
        <div
          className="bg-grey-shd6 w-px"
          style={
            Object {
              "height": "16px",
            }
          }
        />
      </li>
      <li>
        <svg
          data-file-name="SvgHelp"
        />
      </li>
      <li>
        <svg
          data-file-name="SvgHappy"
        />
      </li>
      <li>
        <div
          className="wrapper extra-small"
        >
          <svg
            className="absolute w-full h-full"
            data-file-name="SvgAnonymous"
            height={24}
            width={24}
          />
        </div>
      </li>
    </ul>
  </div>
</nav>
`;
